dot/parachain/availability-distribution/session_cache_test.go

```diff
@@ -109,6 +110,121 @@ func TestReportBadValidators(t *testing.T) {
 	}
 }
 
+func TestGetAuthorityID(t *testing.T) { //nolint:tparallel
+	t.Parallel()
+
+	var (
+		ctrl        *gomock.Controller
+		runtimeMock *MockInstance
+		cache       *LRUSessionCache
+
+		relayParent = common.Hash{0x01}
+	)
+
+	setup := func(t *testing.T) {
+		ctrl = gomock.NewController(t)
+		defer ctrl.Finish()
+
+		runtimeMock = NewMockInstance(ctrl)
+		cache = NewLRUSessionCache(nil)
+	}
+
+	testCases := []struct {
+		description    string
+		validatorIndex parachaintypes.ValidatorIndex
+		setUpRuntime   func(t *testing.T, rt *MockInstance)
+		expectedAuthID parachaintypes.AuthorityDiscoveryID
+		errExpected    bool
+	}{
+		{
+			description:    "session_index_for_child_fails",
+			validatorIndex: 0,
+			setUpRuntime: func(t *testing.T, rt *MockInstance) {
+				rt.EXPECT().
+					ParachainHostSessionIndexForChild().
+					Return(parachaintypes.SessionIndex(0), errors.New("fail")).
+					Times(1)
+			},
+			expectedAuthID: parachaintypes.AuthorityDiscoveryID{},
+			errExpected:    true,
+		},
+		{
+			description:    "session_info_fails",
+			validatorIndex: 0,
+			setUpRuntime: func(t *testing.T, rt *MockInstance) {
+				rt.EXPECT().
+					ParachainHostSessionIndexForChild().
+					Return(parachaintypes.SessionIndex(0), nil).
+					Times(1)
+
+				rt.EXPECT().
+					ParachainHostSessionInfo(gomock.AssignableToTypeOf(parachaintypes.SessionIndex(0))).
+					Return(nil, errors.New("fail")).
+					Times(1)
+			},
+			expectedAuthID: parachaintypes.AuthorityDiscoveryID{},
+			errExpected:    true,
+		},
+		{
+			description:    "invalid_validator_index",
+			validatorIndex: 1337,
+			setUpRuntime: func(t *testing.T, rt *MockInstance) {
+				rt.EXPECT().
+					ParachainHostSessionIndexForChild().
+					Return(parachaintypes.SessionIndex(0), nil).
+					Times(1)
+
+				rt.EXPECT().
+					ParachainHostSessionInfo(gomock.AssignableToTypeOf(parachaintypes.SessionIndex(0))).
+					Return(&parachaintypes.SessionInfo{
+						DiscoveryKeys: []parachaintypes.AuthorityDiscoveryID{{0x01}, {0x02}, {0x03}, {0x04}},
+					}, nil).
+					Times(1)
+			},
+			expectedAuthID: parachaintypes.AuthorityDiscoveryID{},
+			errExpected:    true,
+		},
+		{
+			description:    "happy_path",
+			validatorIndex: 1,
+			setUpRuntime: func(t *testing.T, rt *MockInstance) {
+				rt.EXPECT().
+					ParachainHostSessionIndexForChild().
+					Return(parachaintypes.SessionIndex(0), nil).
+					Times(1)
+
+				rt.EXPECT().
+					ParachainHostSessionInfo(gomock.AssignableToTypeOf(parachaintypes.SessionIndex(0))).
+					Return(&parachaintypes.SessionInfo{
+						DiscoveryKeys: []parachaintypes.AuthorityDiscoveryID{{0x01}, {0x02}, {0x03}, {0x04}},
+					}, nil).
+					Times(1)
+			},
+			expectedAuthID: parachaintypes.AuthorityDiscoveryID{0x02},
+			errExpected:    false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			setup(t)
```

Comment:
Can we run them parallelly?

Reply:
No, because they share variables so that setup() doesn't need to have seven return values. I could change that, but all the tests in dot/parachain/availability-distribution take less than half a second to run on my laptop. If we want to speed up the test suite, we should probably look elsewhere first. :)